### PR TITLE
Bug 1840958: Source can be empty in general step

### PR DESCRIPTION
**Pull request by yaacov**
URL and Container source are not required in the general step, they are part of the storage step.

Screenshots:
Edit storage "rootdisk" URL source:
screenshot-localhost_9000-2020 08 09-15_19_54

Edit storage "rootdisk" Container source:
screenshot-localhost_9000-2020 08 09-15_20_32

Edit storage "rootdisk" "any" source:
screenshot-localhost_9000-2020 08 09-15_21_07

Edit network "any" net interface
screenshot-localhost_9000-2020 08 09-15_29_56

### CODE REVIEW

**Comment by yaacov** [comment]
@matthewcarleton please review

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx [hunk 1]
```diff
@@ -130,6 +194,13 @@ export class VMSettingsTabComponent extends React.Component<VMSettingsTabCompone
               )}
             </FormSelect>
           </FormField>
+          {[
+            ProvisionSource.URL.toString(),
+            ProvisionSource.CONTAINER.toString(),
+            ProvisionSource.DISK.toString(),
+          ].includes(provisionSourceValue) && getProvisionSourceDiskHelpMsg()}
```
Comment (irosenzw) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx:201]: can `getProvisionSourceDiskHelpMsg` be a component rather than a function ?

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx [hunk 2]
```diff
@@ -53,6 +62,61 @@ export class VMSettingsTabComponent extends React.Component<VMSettingsTabCompone
...
+
+    const getProvisionSourceDiskHelpMsg = () => {
+      const storageBtn = (
+        <Button
+          isDisabled={!steps[VMWizardTab.STORAGE]?.canJumpTo}
+          isInline
+          onClick={() => goToStep(VMWizardTab.STORAGE)}
+          variant={ButtonVariant.link}
+        >
+          <strong>storage</strong>
```
Comment (irosenzw) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx:75]: please capitalize

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx [hunk 3]
```diff
@@ -53,6 +62,61 @@ export class VMSettingsTabComponent extends React.Component<VMSettingsTabCompone
...
+
+    const getProvisionSourceNetHelpMsg = () => {
+      const networkBtn = (
+        <Button
+          isDisabled={!steps[VMWizardTab.NETWORKING]?.canJumpTo}
+          isInline
+          onClick={() => goToStep(VMWizardTab.NETWORKING)}
+          variant={ButtonVariant.link}
+        >
+          <strong>networking</strong>
```
Comment (irosenzw) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx:109]: also capitalize here

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx (hunk 2 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx]: +1

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx (hunk 3 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx]: +1

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx (hunk 1 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx]: +1, changed to consts

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts [hunk 4]
```diff
@@ -110,11 +110,9 @@ const provisioningSourceUpdater = ({ id, prevState, dispatch, getState }: Update
   dispatch(
     vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
       [VMSettingsField.CONTAINER_IMAGE]: {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts:112]: the fields were kept required so it would be apparent from redux which fields have to be filled.

for this special use case `skipValidation` was introduced into `vm-settings-tab-initial-state`.

Please revert this change or give a good reason why it shouldn't be visible in redux and skipValidation should then be removed from initial-state?

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx [hunk 5]
```diff
@@ -53,6 +62,52 @@ export class VMSettingsTabComponent extends React.Component<VMSettingsTabCompone
       goToStep,
     } = this.props;
 
+    const provisionSourceValue = this.getFieldValue(VMSettingsField.PROVISION_SOURCE_TYPE);
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx:65]: there are many fields in General Tab and we try to keep this component short as possible. Please either make a new component for this message or make a new component for provision source which will include this functionality.

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx:65]: makes sense to me too :+1:

I can make a follow up PR to move the "source" input into a separate component.

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts (hunk 4 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts:112]: > the fields were kept required so it would be apparent from redux which fields have to be filled.

this fields are are not required in the user view, a user can leave them empty and continue filling up the form.
the redux viewer should reflect that.

> or give a good reason why it shouldn't be visible in redux and skipValidation should then be removed from initial-state?

This fields are optional, users can fill this information later in the storage tab, we should remove the skipValidation, I can make a folow up PR to fix that.

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts:112]: +1 let's remove the skipValidation then

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx (hunk 5 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx:65]: +1
